walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,13 +28,28 @@
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> optionalAttendeesConsideredRanges = new ArrayList<>();
+
+        /* try to remove the timeRanges optional attendees can't make it to. If there's nothing left, ignore the optional attendees*/
+        optionalAttendeesConsideredRanges = availableRanges(events, request, true);
+
+        if (request.getAttendees().isEmpty() ) { // only optional Attendees
```

Comment:
I think the ordering of the logic here is a bit hard to follow: intuitively I would expect that the first check would be:
```
if (!optionalAttendeesConsideredRanges.isEmpty()) {
  return optionalAttendeesConsideredRanges;
}
```
since if there's times available for both required and optional attendees, we should return that.

Reply:
resolved